Code/BlockadeVisualsHandler.cs

```diff
@@ -0,0 +1,199 @@
+    {
+        public static GameObject bcm;
+        public static GameObject poofEffect;
+        public static Dictionary<string, MapData> editedDatas;
+
+        private static GameObject blockadeStand; 
+        private static string bundlePath = BepInEx.Utility.CombinePaths(BepInEx.Paths.PluginPath, PluginInfo.PLUGIN_NAME, "Assets", "apmodels");
+        private static List<BlockadeData> blockadeData;
+        private static Dictionary<string, List<GameObject>> currentSceneBlockades;
+        private static Texture2D lockedPath;
+        private static Texture2D openPath;
+        private static MapIconData blockadeIconData;
+
+        public static void Init()
+        {
+            editedDatas = new();
+
+            if (!ItemRandomizer.Instance.settings.BlockRegionConnections) return;
+
+            string dataPath = BepInEx.Utility.CombinePaths(BepInEx.Paths.PluginPath, PluginInfo.PLUGIN_NAME, "Data", "blockadeData.json");
+            string assetPath = BepInEx.Utility.CombinePaths(BepInEx.Paths.PluginPath, PluginInfo.PLUGIN_NAME, "Assets");
+            if (!ModCore.Utility.TryParseJson<List<BlockadeData>>(dataPath, out blockadeData))
+            {
+                Plugin.Log.LogError("Unable to load Blockade Data! Blockades will not be available.");
+                return;
+            }
+            bcm.GetComponentInChildren<Sign>()._configString = null;
+
+            if (blockadeStand == null)
+            {
+                blockadeStand = GameObject.Instantiate(ModCore.Utility.LoadAssetFromBundle(bundlePath, "Assets/Extra2Dew/Prefabs/AP/Blockade.prefab"));
+                GameObject newSign = GameObject.Instantiate(bcm.GetComponentInChildren<Sign>().gameObject);
+                newSign.transform.SetParent(blockadeStand.transform, false);
+                Object.DontDestroyOnLoad(blockadeStand);
+                blockadeStand.SetActive(false);
+            }
+
+            lockedPath = ModCore.Utility.GetTextureFromFile(assetPath + "/MapIconBlockadeClosed.png");
+            openPath = ModCore.Utility.GetTextureFromFile(assetPath + "/MapIconBlockadeOpen.png");
+
+            CreateBlockadeIconData();
+
+            CreateMapMarkers();
+
+            ItemRandomizer.OnItemReceived += DisableBlockades;
+        }
+
+        private static void CreateBlockadeIconData()
+        {
+            blockadeIconData = ScriptableObject.CreateInstance<MapIconData>();
+            blockadeIconData.name = "BlockadeIconData";
+            blockadeIconData._icon = lockedPath;
+            blockadeIconData._animator = "big";
+            MapMarkerPoint.AltIcon altIcon = new();
+            altIcon.name = "cleared";
+            altIcon.icon = openPath;
+            altIcon.animator = "big";
+            blockadeIconData._altIcons = [altIcon];
+        }
+
+        public static void CreateMapMarkers()
+        {
+            var markerSaver = ModCore.Plugin.MainSaver.GetSaver("/local/markers");
```

Comment:
I'm not going to mandate it, but please try to refrain from using `var` going forward, unless the type is obvious (eg. in case of using a generic method like `List<int>`). By looking at that line, you have no way of knowing it's of type `IDataSaver` without looking at what the `GetSaver()` method returns. And using `var` isn't best practice except in specific situations.

Reply:
Hmm, I wonder what I could possibly be assigning to markerSAVER using a method called GetSAVER, couldn't possibly be a SAVER of some kind.

I'll change it, but the fact it returns a saver is incredibly obvious here.